proposals/0030-core-image-story.md

```diff
@@ -0,0 +1,427 @@
+- Start Date: 2023-21-02
+- Reference Issues: #5202 (and all the issues about `@astrojs/image`, really)
+- Implementation PR: https://github.com/withastro/astro/pull/6344
+
+# Summary
+
+This RFC aims to simplify the usage, optimization and resizing of images in Astro.
+
+# Nomenclature
+
+- **CLS**: Content Layout Shift. A score determining how much content shifted on your page during loading, you want to be as close to 0 (or "no CLS") as possible.
+
+# Example
+
+> Note: All the `src` paths in the examples below represent the final paths at build. During development, those paths may be different due to implementations details.
+
+> Note 2: All the results below, unless specified otherwise, are done using the default image services provided with Astro.
+
+## Image Component
+
+### Image located in `src`, in an Astro file
+
+```astro
+---
+import { Image } from 'astro:assets';
+import myImage from "../assets/my_image.png"; // Image is 1600x900
+---
+
+<Image src={myImage} alt="..." />
+```
+
+#### Result
+
+```html
+<!-- Image is optimized, proper attributes are enforced -->
+<img
+  src="/_astro/my_image.hash.webp"
+  width="1600"
+  height="900"
+  decoding="async"
+  loading="lazy"
+  alt="..."
+/>
+```
+
+### Remote images
+
+```astro
+---
+import { Image } from "astro:assets";
+// Image is 1920x1080
+---
+
+<Image src="https://example.com/image.png" alt="..." />
+<!-- ERROR! `width` and `height` are required -->
+
+<Image src="https://example.com/image.png" width={1280} alt="..." />
+<!-- ERROR! `height` is required -->
+
+<Image src="https://example.com/image.png" width={1280} height={720} alt="..." />
```

Comment:
I wonder if, among these examples, we should show the `alt` prop as mandatory. While it is not required per se, not having `alt` is a considerable downside for accessibility.